_posts/2020-10-30-snowflake-dataOps.md

```diff
@@ -0,0 +1,165 @@
+grant role SYSADMIN to role pipeline_role;
+
+```
+
+Optional:
+In my sql scripts, I am creating tables in my_db database and my_schema schema. So if you want my template scripts run seccessfully on your snowflake account, you will need to run below statements too 
+```
+create database my_db;
+grant all on database my_db to role pipeline_role;
+
+create schema my_schema;
+grant all on schema my_schema to role pipeline_role;
+```
+
+# Step 4: Update pipeline parameters 
+Last step before creating your pipeline is to update parameters with the naming conventions you used to create your snowflake resources. If you have followed my naming convention you probably will not need to change anything except SnowflakeAccount.
+Update both parameter files pipeline/aws_seed-cli-parameters.json and aws_seed.json to match resources you created in snowflake
+
+```
+ "SnowflakeAccount": "<ACCOUNTNAME>.<REGION>",
+ "SnowflakeUsername": "pipeline_sys_user",
+ "SnowflakeMigrationDatabaseName": "pipeline_db_migration_plan",
+ "SnowflakeWarehouse": "pipeline_warehouse",
+ "SnowflakeRole": "pipeline_role",
+```
+
+```
+  {
+    "ParameterKey": "SnowflakeAccount",
+    "ParameterValue": "<ACCOUNTNAME>.<REGION>"
+  },
+  {
+    "ParameterKey": "SnowflakeUsername",
+    "ParameterValue": "pipeline_sys_user"
+  },
+  {
+    "ParameterKey": "SnowflakeWarehouse",
+    "ParameterValue": "pipeline_warehouse"
+  },
+  {
+    "ParameterKey": "SnowflakeMigrationDatabaseName",
+    "ParameterValue": "pipeline_db_migration_plan"
+  },
+  {
+    "ParameterKey": "SnowflakeRole",
+    "ParameterValue": "pipeline_role"
+  }
+```
+
+# Step 4: Deploy your code to AWS
+To deploy the infrastructure for your pipeline, you will need to first setup your aws credentials in your terminal. Next you will need to execute init.sh file
+
+```
+sh pipeline/init.sh
+```
+Make sure all the steps goes through successfully. The above init script executes two major steps:
+1. Creating the cloudformation stack for deploying all the AWS infrastructure including CodePipeline, Codebuild and Codecommit
+2. Pushing your codebase into the newly created Codecommit repository
+
+If the second step fails because of your git config settings, you will need to make sure to run them again
+<center><img src="/img/blog/dataops/init.png" /></center><br/>
+Now you can check codepipeline and see a green end to end deployment
+<left><img src="/img/blog/dataops/codepipeline.png" /></left><br/>
+
+This repo uses Flyway to deploy Snowflake changes. To deploy new changes, all you need is placing your new sql scripts in the sql folder and pushing it to codecommit.
+
+Make sure to follow Flyway the naming conventions. All version sql files should start with V__ and repeatable scripts should start with R___
+<img src="/img/blog/dataops/flyway.png" /><br/>
+
+
+# References:
+
+Flyway: 
+https://flywaydb.org/documentation/database/snowflake
+
+Inception Pipeline:
```

Comment:
I'd remove these links for Inception Pipeline or include only one of them

Reply:
We really should stop using the inception pipeline for small guides - it's an incredible amount of content to consume just to kick the tyres on something else.